### PR TITLE
Modification for Cowell to accommodate New Call Signature 

### PR DESCRIPTION
**Pull request by DhruvSondhi**
Made modifications as mentioned in #1053

### CODE REVIEW

**Comment by DhruvSondhi** [comment]
Just a test right now. More commits coming up:wink:

**Comment by astrojuanlu** [comment]
Thanks! As I clarified in the chat, let's keep this branch only for the refactor. We'll do the switch in another pull request.

File: pyproject.toml [hunk 1]
```diff
@@ -43,6 +43,7 @@ requires = [
     "plotly ~=4.0",
     "pyerfa",
     "scipy >=1.4.0",
+    "numbakit-ode",
```
Comment (astrojuanlu) [inline pyproject.toml]: Please remove this line for now :)

**Comment by DhruvSondhi** [comment]
These tests work fine ... you can check & confirm the same :smile:

File: src/poliastro/twobody/propagation.py [hunk 2]
```diff
@@ -22,10 +22,11 @@
 +-------------+------------+-----------------+-----------------+
 
 """
+import nbkode
```
Comment (astrojuanlu) [inline src/poliastro/twobody/propagation.py]: Not needed yet

File: src/poliastro/twobody/propagation.py [hunk 3]
```diff
@@ -22,10 +22,11 @@
...
 import numpy as np
 from astropy import units as u
 from astropy.coordinates import CartesianDifferential, CartesianRepresentation
-from scipy.integrate import DOP853, solve_ivp
+from scipy.integrate import solve_ivp, DOP853
```
Comment (astrojuanlu) [inline src/poliastro/twobody/propagation.py]: I think this is out of order, `tox -e check` will complain, I think

File: src/poliastro/twobody/propagation.py [hunk 4]
```diff
@@ -86,10 +87,11 @@ def cowell(k, r, v, tofs, rtol=1e-11, *, events=None, f=func_twobody):
     tofs = tofs.to(u.s).value
 
     u0 = np.array([x, y, z, vx, vy, vz])
+    t0 = 0
```
Comment (astrojuanlu) [inline src/poliastro/twobody/propagation.py]: Not needed

File: src/poliastro/twobody/propagation.py [hunk 5]
```diff
@@ -86,10 +87,11 @@ def cowell(k, r, v, tofs, rtol=1e-11, *, events=None, f=func_twobody):
...
 
     result = solve_ivp(
         f,
-        (0, max(tofs)),
+        (0, max(tofs)),
```
Comment (astrojuanlu) [inline src/poliastro/twobody/propagation.py]: Please remove extra space

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Added a bunch of preliminary comments - also, I guess the actual fixes for the tests are yet to come.

This will need a couple of iterations more.

**Comment by astrojuanlu** [comment]
Also, when you finish, please squash the commits into one.

One last question: how are you running the tests? Can you paste the command here?

**Comment by DhruvSondhi** [comment]
I am running the test with the following command: 
```python
pytest test_propagation.py  #running from inside /src/poliastro/tests/
```

**Comment by astrojuanlu** [comment]
In that case, please run

```
$ tox -e reformat  # Rewrites the code to follow our style guides
$ tox -e py38  # assuming you're on Python 3.8
```

This will run _all_ the tests. It will take a while and you will discover more failures.

**Comment by DhruvSondhi** [comment]
> In that case, please run
> 
> ```
> $ tox -e reformat  # Rewrites the code to follow our style guides
> $ tox -e py38  # assuming you're on Python 3.8
> ```
> 
> This will run _all_ the tests. It will take a while and you will discover more failures.

the `py38` here is the `environment` ?

**Comment by astrojuanlu** [comment]
yep!

**Comment by astrojuanlu** [comment]
...the tox environment. Not the Python virtual environment.

**Comment by astrojuanlu** [comment]
Merged manually in 640aebbd324467af5ffffd4a6aa7c89b36e2756a. Thanks @DhruvSondhi!

**Pull request closed without merging**